langchain/vectorstores/redisearch.py

```diff
@@ -0,0 +1,192 @@
+from __future__ import annotations
+
+import json
+import uuid
+from typing import Any, Callable, Iterable, List, Mapping, Optional
+
+import numpy as np
+from redis.commands.search.query import Query
+
+from langchain.docstore.document import Document
+from langchain.embeddings.base import Embeddings
+from langchain.utils import get_from_dict_or_env
+from langchain.vectorstores.base import VectorStore
+
+
+class RediSearch(VectorStore):
+    def __init__(
+        self, redisearch_url: str, index_name: str, embedding_function: Callable, **kwargs: Any
+    ):
+        """Initialize with necessary components."""
+        try:
+            import redis
+        except ImportError:
+            raise ValueError(
+                "Could not import redis python package. "
+                "Please install it with `pip install redis`."
+            )
+        self.embedding_function = embedding_function
+        self.index_name = index_name
+        try:
+            redis_client = redis.from_url(redisearch_url, **kwargs)
+        except ValueError as e:
+            raise ValueError(f"Your redis connected error: {e}")
+        self.client = redis_client
+
+    def add_texts(
+        self,
+        texts: Iterable[str],
+        metadatas: Optional[List[dict]] = None,
+        **kwargs: Any,
+    ) -> List[str]:
+        # `prefix`: Maybe in the future we can let the user choose the index_name.
+        prefix = "doc"  # prefix for the document keys
+
+        ids = []
+        # Check if index exists
+        for i, text in enumerate(texts):
+            key = f"{prefix}:{uuid.uuid4().hex}"
+            metadata = metadatas[i] if metadatas else {}
+            self.client.hset(
+                key,
+                mapping={
+                    "content": text,
+                    "content_vector": np.array(
+                        self.embedding_function(text), dtype=np.float32
+                    ).tobytes(),
+                    "metadata": json.dumps(metadata),
+                },
+            )
+            ids.append(key)
+        return ids
+
+    def similarity_search(
+        self, query: str, k: int = 4, **kwargs: Any
+    ) -> List[Document]:
+        # Creates embedding vector from user query
+        embedding = self.embedding_function(query)
+
+        # Prepare the Query
+        return_fields = ["metadata", "content", "vector_score"]
+        vector_field = "content_vector"
+        hybrid_fields = "*"
+        base_query = (
+            f"{hybrid_fields}=>[KNN {k} @{vector_field} $vector AS vector_score]"
+        )
+        redis_query = (
+            Query(base_query)
+            .return_fields(*return_fields)
+            .sort_by("vector_score")
+            .paging(0, k)
+            .dialect(2)
+        )
+        params_dict: Mapping[str, str] = {
+            "vector": str(np.array(embedding).astype(dtype=np.float32).tobytes())
```

Comment:
Hi @xinqiu this is critical -- this vector should not be cast as `str` here otherwise it fails the integration test. Please remove the `str()` wrapper.

Reply:
I used `str()` just wanna to fix lint
```bash
$ poetry run mypy .      
                                                                                                                                            
langchain/vectorstores/redisearch.py:91: error: Dict entry 0 has incompatible type "str": "bytes"; expected "str": "str"  [dict-item]
Found 1 error in 1 file (checked 365 source files)
```